hw/system/occamy/Makefile

```diff
@@ -11,39 +11,84 @@ ROOT        := ${MKFILE_DIR}../../..
 
 include $(ROOT)/util/Makefrag
 
-VLT_FLAGS 	  += -Wno-SYMRSVDWORD
 REGGEN        ?= ${ROOT}/util/regtool.py
-CLANG_FORMAT  ?= $(shell which clang-format-10.0.1)
+OCCAMYGEN     ?= ${ROOT}/util/occamygen.py
 
-QUESTA_64BIT	= -64
-VSIM      		= vsim ${QUESTA_64BIT}
-VLOG      		= vlog ${QUESTA_64BIT}
+CLANG_FORMAT  ?= $(shell which clang-format-10.0.1)
 
-VSIM_FLAGS    += -t 1ps
-VSIM_FLAGS    += -voptargs=+acc
-VSIM_FLAGS    += -do "log -r /*; run -a"
+VSIM      		= vsim -64
+VLOG      		= vlog -64
+
+TB_CC_SOURCES += \
+	$(TB_DIR)/rtl_lib.cc \
+	$(TB_DIR)/common_lib.cc \
+	$(ROOT)/hw/system/occamy/test/bootdata.cc \
+	$(ROOT)/hw/system/occamy/test/uartdpi/uartdpi.c
+
+TB_CC_FLAGS += \
+	-std=c++14 \
+	-I$(MKFILE_DIR) \
+	-I$(MKFILE_DIR)/test \
+	-I$(FESVR)/include \
+	-I$(TB_DIR) \
+	-I$(ROOT)/hw/system/occamy/test/uartdpi
+
+# Required C sources for the verilator TB that are linked against the verilated model
+VLT_COBJ += \
+    $(VLT_BUILDDIR)/tb/common_lib.o \
+	$(VLT_BUILDDIR)/tb/verilator_lib.o \
+	$(VLT_BUILDDIR)/tb/tb_bin.o \
+	$(VLT_BUILDDIR)/test/uartdpi/uartdpi.o \
+	$(VLT_BUILDDIR)/test/bootdata.o
+# Sources from verilator root
+VLT_COBJ += \
+	$(VLT_BUILDDIR)/vlt/verilated.o \
+	$(VLT_BUILDDIR)/vlt/verilated_dpi.o \
+	$(VLT_BUILDDIR)/vlt/verilated_vcd_c.o
+
+VSIM_FLAGS += -t 1ps
+VSIM_FLAGS += -voptargs=+acc
+VSIM_FLAGS += -do "log -r /*; run -a"
 # Downgraded from error as VSIM *wrongly* reports illegal enum assignments in riscv_dbg 4-phase CDCs
 # TODO: Workaround; revise
-VSIM_FLAGS    += -warning 8386
+VSIM_FLAGS += -warning 8386
+
+VCS_FLAGS += +nospecify
+VCS_FLAGS += -Mdir=work-vcs
+VCS_FLAGS += -Mlib=work-vcs
+VCS_FLAGS += -full64
+# Multi-core
+VCS_FLAGS += -fgp
+# Debug Options
+VCS_FLAGS += -assert disable_cover
+VCS_FLAGS += -kdb
+VCS_FLAGS += +vcs+fsdbon
+VCS_FLAGS += +lint=TFIPC-L
+VCS_FLAGS += -debug_access+all
+
+VLOG_FLAGS += -svinputport=compat
+VLOG_FLAGS += -override_timescale 1ns/1ps
+VLOG_FLAGS += -suppress 2583
+VLOG_FLAGS += -suppress 13314
+VLOG_FLAGS += ${QUESTA_64BIT}
+VLOG_FLAGS += -timescale 1ns/1ps
+
+# Verilated and compiled Occamy system
+VLT_AR     = ${VLT_BUILDDIR}/Vtestharness__ALL.a
+VLT_FLAGS += -Wno-SYMRSVDWORD -Wno-BLKLOOPINIT
+VLT_FLAGS += --trace       # Enable VCD dumping
+VLT_FLAGS += --prof-cfuncs # Enable code profiling
```

Comment:
Do thes increase the default runtime? If so we might want to be careful with adding them by default. Hints for some useful flags could be added in some documentation/readme.